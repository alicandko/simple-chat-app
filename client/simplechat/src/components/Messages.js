import React from 'react';
import moment from 'moment';

export const Messages = ({ messages }) => {
  const messageElements = messages.map((message, index) => (
    <div className="messages__message" key={message.id}>
      <div className="messages__username">{message.username}</div>
      <div className="messages__text">{message.text}</div>
      <div className="messages__date">{moment(message.date).format('Do MMM YYYY h:mm')}</div>
    </div>
  ));

  return <div className="messages">{messageElements}</div>;
};
