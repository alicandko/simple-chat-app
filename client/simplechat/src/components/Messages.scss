.messages {
  border: 1px solid #f2f2f2;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  margin: 0;
  padding: 10px 20px;
  width: 100%;
  display: flex;
  flex-direction: column;

  &__message {
    display: flex;
    flex-direction: column;
    margin-bottom: 10px;
  }

  &__username,
  &__date {
    line-height: 20px;
    opacity: 0.3;
  }

  &__text {
    line-height: 20px;
  }
}
