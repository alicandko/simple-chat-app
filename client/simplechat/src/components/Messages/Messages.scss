.messages {
  border: 1px solid #adc5e6;
  border-bottom: none;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  margin: 0;
  padding: 10px 20px;
  width: 100%;
  min-height: 113px;
  display: flex;
  flex-direction: column;

  &__message {
    display: flex;
    flex-direction: column;
    margin-bottom: 10px;

    &--you {
      border: 1px solid #adc5e6;
      border-radius: 3px;
      padding: 10px;
    }
  }

  &__username,
  &__date {
    line-height: 20px;
    opacity: 0.5;
  }

  &__text {
    line-height: 20px;
  }
}
