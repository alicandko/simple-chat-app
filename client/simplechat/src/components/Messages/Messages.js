import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import './Messages.scss';

const propTypes = {
  username: PropTypes.string.isRequired,
  messages: PropTypes.arrayOf(Object).isRequired,
};

const YOU = 'You';
const DATE_FORMAT = 'Do MMM YYYY h:mm';

export const Messages = ({ username, messages }) => {
  const messageElements = messages.map((message) => {
    const { id, username: messageUsername, text, date } = message;
    const messageClassModifier = messageUsername === username ? 'messages__message--you' : '';

    return (
      <div className={'messages__message ' + messageClassModifier} key={id}>
        <div className="messages__username">
          {messageUsername === username ? YOU : messageUsername}
        </div>
        <div className="messages__text">{text}</div>
        <div className="messages__date">{moment(date).format(DATE_FORMAT)}</div>
      </div>
    );
  });

  return <div className="messages">{messageElements}</div>;
};

Messages.propTypes = propTypes;
