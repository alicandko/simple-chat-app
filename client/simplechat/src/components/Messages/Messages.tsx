import moment from 'moment';
import React, { FunctionComponent } from 'react';
import { IncomingMessage } from '../../redux/types';

import './Messages.scss';

interface Props {
  messages: IncomingMessage[];
  username: string;
}

const YOU = 'You';
const DATE_FORMAT = 'Do MMM YYYY HH:mm';

export const Messages: FunctionComponent<Props> = ({ username, messages }) => {
  const messageElements = messages.map((message) => {
    const { id, username: messageUsername, text, date } = message;
    const messageClassModifier = messageUsername === username ? 'messages__message--you' : '';

    return (
      <div className={'messages__message ' + messageClassModifier} key={id}>
        <div className="messages__username">
          {messageUsername === username ? YOU : messageUsername}
        </div>
        <div className="messages__text">{text}</div>
        <div className="messages__date">{moment(date).format(DATE_FORMAT)}</div>
      </div>
    );
  });

  return <div className="messages">{messageElements}</div>;
};
