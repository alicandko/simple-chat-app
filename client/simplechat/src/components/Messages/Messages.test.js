import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Messages } from './Messages';

const mockMessages = [
  { id: 1, date: '2019-06-05T18:10:42.954+0000', text: 'mockText', username: 'mockUsername' },
  {
    id: 2,
    date: '2019-06-06T18:10:42.954+0000',
    text: 'anotherMockText',
    username: 'anotherMockUsername',
  },
];

describe('MessageForm', () => {
  it('should render correctly', () => {
    const component = renderer.create(<Messages username="" messages={[]} />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should have the correct passed props', () => {
    const wrapper = mount(<Messages username="mockUsername" messages={mockMessages} />);

    expect(wrapper.prop('username')).toEqual('mockUsername');
    expect(wrapper.prop('messages')).toEqual(mockMessages);
  });

  it('should render correct number of messages', () => {
    const wrapper = mount(<Messages username="mockUsername" messages={mockMessages} />);

    expect(wrapper.find('.messages__message')).toHaveLength(2);
  });

  it('should render username, text and date of a message', () => {
    const wrapper = mount(<Messages username="mockUsername" messages={mockMessages} />);
    const messageWrapper = wrapper.find('.messages__message').first();

    expect(messageWrapper.find('.messages__username')).toHaveLength(1);
    expect(messageWrapper.find('.messages__text')).toHaveLength(1);
    expect(messageWrapper.find('.messages__date')).toHaveLength(1);
  });
});
