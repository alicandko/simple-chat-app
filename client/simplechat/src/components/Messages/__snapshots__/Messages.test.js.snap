// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm should render correctly 1`] = `
<div
  className="messages"
/>
`;
